Packs/ApiModules/Scripts/GetIncidentsApiModule/GetIncidentsApiModule.py

```diff
@@ -0,0 +1,228 @@
+from CommonServerPython import *
+
+DEFAULT_LIMIT = 500
+DEFAULT_PAGE_SIZE = 100
+DEFAULT_TIME_FIELD = "created"
+
+
+def build_query_parameter(
+    custom_query: str | None,
+    incident_types: list[str],
+    time_field: str,
+    from_date: str | None,
+    to_date: str | None,
+    non_empty_fields: list[str],
+) -> str:
+    """Builds the query parameter string from given arguments.
+
+    Args:
+        custom_query (str | None): A custom query.
+        incident_types (list[str] | None): Incident types to retrieve.
+        time_field (str): The relevant time field to search by.
+        from_date (str | None): The start date for the incidents query.
+        to_date (str | None): The end date for the incidents query.
+        non_empty_fields (list[str]): Required non-empty incident fields.
+
+    Raises:
+        Exception: If no query parts were added.
+
+    Returns:
+        str: The query string built from the given arguments.
+    """
+    query_parts = []
+    if custom_query:
+        query_parts.append(custom_query)
+    if incident_types:
+        types = [x if "*" in x else f'"{x}"' for x in incident_types]
+        query_parts.append(f"type:({' '.join(types)})")
+    if from_date and time_field:
+        query_parts.append(f'{time_field}:>="{from_date}"')
+    if to_date and time_field:
+        query_parts.append(f'{time_field}:<"{to_date}"')
+    if non_empty_fields:
+        query_parts.append(" and ".join(f"{x}:*" for x in non_empty_fields))
+    if not query_parts:
+        raise DemistoException("Incidents query is empty - please fill at least one argument")
+    return " and ".join(f"({x})" for x in query_parts)
+
+
+def format_incident(inc: dict, fields_to_populate: list[str], include_context: bool) -> dict:
+    """Flattens custom fields with incident data and filters by fields_to_populate
+
+    Args:
+        inc (dict): An incident.
+        fields_to_populate (list[str]): List of fields to populate.
+        include_context (bool): Whether or not to enrich the incident with its context data.
+
+    Returns:
+        dict: The formatted incident.
+    """
+    custom_fields = inc.pop('CustomFields', {})
+    inc.update(custom_fields)
+    if fields_to_populate:
+        inc = {k: v for k, v in inc.items() if k.lower() in {val.lower() for val in fields_to_populate}}
+        if any(f.lower() == "customfields" for f in fields_to_populate):
+            inc["CustomFields"] = custom_fields
```

Comment:
this was added to support providing `CustomFields` as a field to populate, which allows using it in the `similarJsonField` argument of the `DBotFindSimilarIncidents` script, as mentioned in its description.